denylist/src/denylist.rs

```diff
@@ -122,21 +123,21 @@ pub fn filter_from_bin(bin: &Vec<u8>) -> Result<Xor32> {
     let _version = buf.get_u8();
     let signature_len = buf.get_u16_le() as usize;
     let signature = buf.copy_to_bytes(signature_len).to_vec();
-    let pubkey = PublicKey::from_str(PUB_KEY_B58)?;
-    match pubkey.verify(buf, &signature) {
-        Ok(_) => {
-            tracing::info!("updating filter to latest");
-            let _serial = buf.get_u32_le();
-            let xor = bincode::deserialize::<Xor32>(buf)?;
-            Ok(xor)
-        }
-        Err(_) => {
-            tracing::warn!("filter signature verification failed");
-            Err(Error::InvalidBinary(
-                "filter signature verification failed".to_string(),
-            ))
+    for pubkey in valid_sign_keys {
+        match pubkey.verify(buf, &signature) {
+            Ok(_) => {
+                tracing::info!("updating filter to latest");
+                let _serial = buf.get_u32_le();
+                let xor = bincode::deserialize::<Xor32>(buf)?;
+                return Ok(xor);
+            }
+            Err(_) => continue,
```

Comment:
sign_keys
    .iter()
    .any(|key| key.verify(buf, &signature).is_ok())
    .then(|| {
        let _serial = buf.get_u32_le();
        bincode::deserialize::<Xor32>(buf)
    })
    .transpose()?
    .ok_or_else(|| {
        tracing::warn!("filter signature verification failed");
        Err(Error::InvalidBinary("filter signature verification_failed".to_string()))
    })

Reply:
^ this is much better IMO